code-challenges.test.js

```diff
@@ -60,8 +186,42 @@ var hand2 = [5, 5, 3, 3, 4]
 var hand3 = [5, 5, 5, 5, 4]
 // Expected output: false
 
-
-
-
-
 // b) Create the function that makes the test pass.
+
+// create a function that is called determineFullHouse
+// it takes in an array an returns a boolean
+// if theres 3 of integer and 2 of something then full house is true
+// 0 and compare iterating thought the array and checking against the character looking at the 
+// find two unique values and check to see if there is three of one two of the other
+const determineFullHouse = (arr) => {
+    //set the boolean to false as the conditions for full house have not been met yet
+    let fullHouse = false;
+    //create two storage variables
+    let isPair = [];
+    let isTriple = [];
+    //need to iterate though the array... looks like any higher order function would work.. tested with .map and .filter all worked
+    arr.forEach(value =>{
+        //if the isTriple is empty put the first value of the array into isTriple 
+        //then use the first value to match up with all the other values in the array
+        //set condition to limit the array to 3 values once the array reaches the length of 3 then it moves on to next array
+        //do the same that was done to isTriple
+        if (isTriple.length === 0){
+            isTriple.push(value)
+        }else if(isTriple[0] === value && isTriple.length < 3){
+            isTriple.push(value)
+        }else if(isPair.length === 0){
+            isPair.push(value)
+        }else if (isPair[0] === value){
+            isPair.push(value)
+    }})     
+    //set a condition that checks to see the length of each array is 3 and 2 to indicate the number of duplicates
+    //once condition is met fullhouse is set to true.
+    if (isTriple.length === 3 && isPair.length === 2){
+        fullHouse = true
+    }
+// my code was flooded with console logs on this one but these two ended up being very helpful in the end.
+// console.log(isPair)
+// console.log(isTriple)
+return fullHouse
```

Comment:
if you're looking for a challenge, consider how you can ensure that the inputs fit the requirements of the function (e.g. what should happen if someone tried to call your function with [3,3,3,3,2,2,2]?)  This is a test case that I think your function would return true for. Another challenge could also be consider how you could solve this problem using an object oriented approach